src/wallet/rpcwallet.cpp

```diff
@@ -2359,3 +2361,293 @@ UniValue listunspent(const UniValue& params, bool fHelp)
+                            );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    std::vector<HDChainID> chainIDs;
+    if (!pwalletMain->GetAvailableChainIDs(chainIDs))
+        throw runtime_error("Could not load chain ids");
+
+    UniValue result(UniValue::VARR);
+    BOOST_FOREACH(const HDChainID& chainId, chainIDs)
+    {
+        CHDChain chain;
+        if (!pwalletMain->GetChain(chainId, chain))
+            throw runtime_error("Could not load chain");
+
+        UniValue chainObject(UniValue::VOBJ);
+        chainObject.push_back(Pair("chainid", chainId.GetHex()));
+        chainObject.push_back(Pair("creationtime", chain.nCreateTime));
+        chainObject.push_back(Pair("chainpath", chain.chainPath));
+
+        result.push_back(chainObject);
+    }
+
+    return result;
+}
+
+UniValue hdgetaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() > 1)
+        throw runtime_error(
+                            "hdgetaddress (<childindex>)\n"
+                            "\nReturns a Bitcoin address for receiving payments.\n"
+                            "\nAutomatically uses the next available childindex if no index is given"
+                            "\nArguments:\n"
+                            "1. \"childindex\"        (numeric, optional) child key index. ATTENTION: automatic index counting will start at the highes available child key index\n"
+                            "{\n"
+                            "  \"address\" : \"<address>\",  string) The new bitcoin address\n"
+                            "  \"chainpath\" : \"<keyschainpath>\",  string) The used chainpath\n"
+                            "}\n"
+                            "\nExamples:\n"
+                            + HelpExampleCli("hdgetaddress", "")
+                            + HelpExampleCli("hdgetaddress", "100")
+                            + HelpExampleRpc("hdgetaddress", "")
+                            );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    CPubKey newKey;
+    std::string keyChainPath;
+    if (params.size() == 1 && params[0].isNum())
+    {
+        HDChainID emptyId;
+        if (!pwalletMain->HDGetChildPubKeyAtIndex(emptyId, newKey, params[0].get_int()))
+            throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, "Error: Can't generate HD child key");
+    }
+    else
+    {
+        HDChainID emptyId;
+        if (!pwalletMain->HDGetNextChildPubKey(emptyId, newKey, keyChainPath))
+            throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, "Error: Can't generate HD child key");
+    }
+    CKeyID keyID = newKey.GetID();
+    
+    pwalletMain->SetAddressBook(keyID, "", "receive");
+
+    UniValue result(UniValue::VOBJ);
+    result.push_back(Pair("address", CBitcoinAddress(keyID).ToString()));
+    result.push_back(Pair("chainpath", keyChainPath));
+    return result;
+}
+
+UniValue hdsendtoaddress(const UniValue& params, bool fHelp)
```

Comment:
I agree with @luke-jr. The hd'ness should be used by default in future. But maybe to give a save transition, a first version would keep the hd functions separated. Users could decide where they want to use hd and where not. Maybe not everyone likes to use hd change keys for current non-hd inputs.

But right, this PR would slightly mix hd and non-hdness anyway. Unspents and balance would be shared.